lib/membrane_live/event.ex

```diff
@@ -216,18 +226,25 @@ defmodule MembraneLive.Event do
   end
 
   @impl true
-  def handle_info({:playlist_playable, :audio, _playlist_idl}, state) do
+  def handle_info({:playlist_playable, :audio, _playlist_idl, _peer_id}, state) do
     # TODO: implement detecting when HLS starts
     {:noreply, state}
   end
 
   @impl true
-  def handle_info({:playlist_playable, :video, playlist_idl}, state) do
-    state = put_in(state, [:is_playlist_playable], true)
-    state = put_in(state, [:playlist_idl], playlist_idl)
+  def handle_info({:playlist_playable, :video, playlist_idl, peer_id}, state) do
+    state =
+      Map.put(
+        state,
+        :playlist_idls,
+        Enum.map(state.playlist_idls, &add_playlist_idl_to_peer(&1, playlist_idl, peer_id))
+      )
+
+    name = Enum.find(state.playlist_idls, fn map -> map[:playlist_idl] == playlist_idl end).name
```

Comment:
Why don't we simply store under in `state.playlist_idls` a map `%{ peer_id => {peer_name, playlist_idl}}`? For deciding which stream should be played now, we could simply use a queue or stack of `peer_ids`.